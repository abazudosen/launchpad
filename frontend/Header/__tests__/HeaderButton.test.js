/* @flow */

import React from 'react';
import HeaderButton from '../HeaderButton';
import renderer from 'react-test-renderer';

test('renders normal', () => {
  let component = renderer.create(
    <HeaderButton onClick={() => {}}>
      <div>Content</div>
    </HeaderButton>,
  );
  expect(component.toJSON()).toMatchSnapshot();

  component = renderer.create(
    <HeaderButton active onClick={() => {}}>
      <div>Content</div>
    </HeaderButton>,
  );
  expect(component.toJSON()).toMatchSnapshot();

  component = renderer.create(
    <HeaderButton disabled onClick={() => {}}>
      <div>Content</div>
    </HeaderButton>,
  );
  expect(component.toJSON()).toMatchSnapshot();

  component = renderer.create(
    <HeaderButton tooltip={<div>Tooltip</div>} onClick={() => {}}>
      <div>Content</div>
    </HeaderButton>,
  );
  expect(component.toJSON()).toMatchSnapshot();
});
