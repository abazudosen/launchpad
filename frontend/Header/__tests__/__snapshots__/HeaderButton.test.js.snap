// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders normal 1`] = `
<button
  className="HeaderButton"
  onClick={[Function]}
>
  <div
    className="HeaderButton-Inner"
  >
    <div>
      Content
    </div>
  </div>
</button>
`;

exports[`renders normal 2`] = `
<button
  className="HeaderButton HeaderButton--active"
  onClick={[Function]}
>
  <div
    className="HeaderButton-Inner"
  >
    <div>
      Content
    </div>
  </div>
</button>
`;

exports[`renders normal 3`] = `
<button
  className="HeaderButton HeaderButton--disabled"
  onClick={[Function]}
>
  <div
    className="HeaderButton-Inner"
  >
    <div>
      Content
    </div>
  </div>
</button>
`;

exports[`renders normal 4`] = `
<button
  className="HeaderButton"
  onClick={[Function]}
>
  <div
    className="HeaderButton-Inner"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div>
      Content
    </div>
  </div>
  <div
    className="HeaderButton-Tooltip"
  >
    <div>
      Tooltip
    </div>
  </div>
</button>
`;
